### PR TITLE
Rankings and responsive design for player profile and rankings

### PR DESCRIPTION
**Pull request by Alex-git-code**
Rankings and responsive design for player profile and rankings
Add player name restrictions, max 15 characters
Toast message for error and update player name

### CODE REVIEW

File: app/src/main/java/org/alexcode/snake/EditNameDialog.java [hunk 1]
```diff
@@ -51,9 +52,14 @@ public void onClick(DialogInterface dialogInterface, int i) {
                     @Override
                     public void onClick(DialogInterface dialogInterface, int i) {
                         newPlayerName = editTextPlayerName.getText().toString();
-                        playerId = PlayerPreferences.getPlayerId();
-                        if(!newPlayerName.equals(playerName)) {
-                            updatePlayerName();
+                        if(newPlayerName.length() > 15) {
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/EditNameDialog.java:55]:
```suggestion
                        if (newPlayerName.length() > 15) {
```
let's be consistents

File: app/src/main/java/org/alexcode/snake/MainActivity.java [hunk 2]
```diff
@@ -25,7 +25,7 @@ protected void onCreate(Bundle savedInstanceState) {
         playerPreferences =  new PlayerPreferences(this);
         initGameMenu();
         initPlayerData();
-        startGame.setOnClickListener(new View.OnClickListener() {
+        startGame.setOnClickListener(new View.OnClickListener() {
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/MainActivity.java:28]: I think you should get rid of those spaces

File: app/src/main/java/org/alexcode/snake/EditNameDialog.java [hunk 3]
```diff
@@ -51,9 +52,14 @@ public void onClick(DialogInterface dialogInterface, int i) {
                     @Override
                     public void onClick(DialogInterface dialogInterface, int i) {
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/EditNameDialog.java:53]: onClick on what exactly? is not clear enough

File: app/src/main/java/org/alexcode/snake/Rankings.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import android.view.View;
+import android.widget.Button;
+import com.android.volley.Cache;
+import com.android.volley.Network;
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.toolbox.BasicNetwork;
+import com.android.volley.toolbox.DiskBasedCache;
+import com.android.volley.toolbox.HurlStack;
+import com.android.volley.toolbox.StringRequest;
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/Rankings.java:18]: can't you import 'com.android.volley.*' ? there are too many packages included here
same for the org.json libraries above

File: app/src/main/java/org/alexcode/snake/Rankings.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+public class Rankings extends AppCompatActivity {
+    private static final String URL_PLAYERS = "https://upgreat.ro/snake_get_rankings.php";
+    Button btnBackToMenu;
+    //list to store all players
+    List<Player> playersList;
+    //the recyclerview
+    RecyclerView recyclerView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/Rankings.java:34]: I think it should be called onCreateRanking

File: app/src/main/java/org/alexcode/snake/Rankings.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+                        //creating adapter object and setting it to recyclerview
+                        PlayerAdapter adapter = new PlayerAdapter(Rankings.this, playersList);
+                        recyclerView.setAdapter(adapter);
+                    } catch (JSONException e) {
+                        Log.d("GET RANKINGS", "Json error. The error is: " + e);
+                    }
+                },
+                error -> {
+                    Log.d("GET RANKINGS", "Api Call fail.");
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/Rankings.java:91]: this logic should be extracted in another function that dos the http connection and the json extraction

**Review by andreisamoila74 — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/alexcode/snake/EditNameDialog.java (hunk 3 above)
Comment (Alex-git-code) [reply app/src/main/java/org/alexcode/snake/EditNameDialog.java:53]: On click positive button of the dialog box. Check line 51

File: app/src/main/java/org/alexcode/snake/Rankings.java (hunk 4 above)
Comment (Alex-git-code) [reply app/src/main/java/org/alexcode/snake/Rankings.java:18]: Ok

File: app/src/main/java/org/alexcode/snake/MainActivity.java (hunk 2 above)
Comment (Alex-git-code) [reply app/src/main/java/org/alexcode/snake/MainActivity.java:28]: ok

File: app/src/main/java/org/alexcode/snake/EditNameDialog.java (hunk 1 above)
Comment (Alex-git-code) [reply app/src/main/java/org/alexcode/snake/EditNameDialog.java:55]: ok

File: app/src/main/java/org/alexcode/snake/Rankings.java (hunk 5 above)
Comment (Alex-git-code) [reply app/src/main/java/org/alexcode/snake/Rankings.java:34]: I use a predefined method here and override, cannot be renamed

**Pull request merged**